### ISSUE TITLE
Bug: Multiple errors from different lines incorrectly reported on the same line

### ISSUE THREAD

**Issue by ktanaka101** [google/dart-neats#265]
First of all, thank you for creating this amazing library! dartdoc_test is incredibly helpful for maintaining documentation quality.

I found a bug where errors from different lines are being reported as if they occurred on the same line.
This should either be reported on their respective lines or consolidated into a single error message.

### Example case (missing `;`):

```dart
/// This class is Foo.
/// ```dart
/// print('foo')
///
/// print('bar')
/// ```
class FooClass {}
```

### Current output (`dart run dartdoc_test`):

```
Extracting code samples ...
Analyzing code samples ...
src/main.dart:3:16: : Expected to find ';'.
  ╷
3 │ /// print('foo')
  │                ^
  ╵

src/main.dart:3:16: : Expected to find ';'.
  ╷
3 │ /// print('foo')
  │                ^
  ╵

FAILED: 2 issues found (Found 1 code samples in 1 files)
```

### Expected behavior (Option 1 - Report on respective lines):

```
src/main.dart:3:16: : Expected to find ';'.
   ╷
 3 │ /// print('foo')
   │                ^
   ╵

src/main.dart:5:16: : Expected to find ';'.
   ╷
 5 │ /// print('bar')
   │                ^
   ╵

FAILED: 2 issues found (Found 1 code samples in 1 files)
```

### Expected behavior (Option 2 - Consolidated message):

```
src/main.dart:3:16: : Expected to find ';'.
   ╷
 3 │ /// print('foo')
   │                ^
   ╵

FAILED: 1 issue found (Found 1 code samples in 1 files)
```

### Environment:

Dart SDK version: v3.5.4
dartdoc_test version: v0.1.0

**Comment by yash01699** [comment]
Hi @jonasfj , can i take this up if this hasnt been worked on already?

**Comment by jonasfj** [comment]
I'd suggest taking an overall look at `dartdoc_test` given the new `/// {@example /path/to/file.dart}` dartdoc directive, I suspect this package is better discontinued.

See: https://github.com/dart-lang/dartdoc/blob/main/doc/directives.md#example

**Comment by yash01699** [comment]
you're right, this can be discontinued given the new @example directive. Thanks! would love to contribute in any other way possible

**Comment by jonasfj** [comment]
I marked the package as discontinued.

I just filed:
 * https://github.com/google/dart-neats/issues/390
 * https://github.com/google/dart-neats/issues/391

I'd contributions on those, but no LLM generated documentation or examples, that's entirely uninteresting. It's really an art to try and keep documentation concise, while ensure that caveats are covered.
